datamigration/doc.go

```diff
@@ -0,0 +1,27 @@
+/*
+
+Data migration package implements a functionality to declare and execute data
+migrations.
+
+Each data migration is a function registered under a unique name, that can be
+executed only once.
+
+In order to register a new data migration:
+1. register a new migration using `datamigration.MustRegister`,
+2. build a new binary of your application and make sure all validators are running it,
+3. send a transaction with ExecuteMigrationMsg, signed by required addresses,
+4. if migration was executed, it became a part of the chain. Its code must
```

Comment:
What it means here **it became a part of the chain**? Do you mean `migration result becomes persistent in the chain and irreversible`? And what do you mean by **its code must remain unchanged forever**?

Reply:
> it became a part of the chain

Blockchain is a data structure. In a way, it is a memory of a database. Maybe this is too big of a mind jump.
The important part is that migration is persisted on the chain. Maybe that is what I should write there.

> its code must remain unchanged forever

Exactly what I wrote :) You must never change or delete the code of a migration.